### PR TITLE
Implement APM Agent Middleware

### PR DESCRIPTION
**Pull request by reynldi**
### What does this MR do?
Implement APM Agent Middleware

### Author's Checklist
- [x] Follow the Documentation Guidelines
- [x] Use label properly
- [x] Assign to Reviewer

### New Dependency
_put dependency link_

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -2,7 +2,12 @@ import { Module, MiddlewareConsumer, RequestMethod } from '@nestjs/common';
 import i18n from 'i18n';
 
 import Database from '@app/config/database';
+
+/**
+ * Import Middleware
+ */
 import { HeaderMiddleware } from '@app/middleware/header.middleware';
+import APMAgent from '@app/middleware/apm-agent.middleware';
```
Comment (tahsinature) [inline src/app.module.ts]: it's not a class anymore.
it should be `apmAgent`

File: src/app.module.ts [hunk 2]
```diff
@@ -21,6 +26,7 @@ import { CheckEmailService } from './services/check-email.service';
 })
 export class AppModule {
   configure(consumer: MiddlewareConsumer) {
+    consumer.apply(APMAgent.init());
```
Comment (tahsinature) [inline src/app.module.ts]: should connect the middleware like this
`apm.agent.middleware.connect()`

**Review by tahsinature — CHANGES_REQUESTED** [review]
apm middleware integration should be changed

File: src/app.module.ts (hunk 1 above)
Comment (reynldi) [reply src/app.module.ts]: Yah, it just naming right ?

File: src/app.module.ts (hunk 2 above)
Comment (reynldi) [reply src/app.module.ts]: Ok i see .thanks https://www.elastic.co/guide/en/apm/agent/nodejs/master/agent-api.html#apm-middleware-connect

**Pull request merged**